worker/src/main/scala/com/lucidchart/piezo/ConnectionProvider.scala

```diff
@@ -113,7 +111,7 @@ class ConnectionProvider(props: Properties) {
       val newIP: String = getIP
       if (hasIpAddressChanged(pool, newIP)) {
         // A failover has occurred, so we evict connections softly. New connectons look up the new IP address
-        pool.connectionProvider.get.getDataSource().getHikariPoolMXBean().softEvictConnections()
+        pool.connectionProvider.map(_.getDataSource().getHikariPoolMXBean().softEvictConnections())
       }
     }
     pool.connectionProvider.get.getConnection
```

Comment:
You've got another `get` here to handle

Reply:
This is already the behavior before my changes. I'm not too concerned about the `get` here, because some sort of exception needs to be thrown if the `connectionProvider` doesn't exist when trying to get a connection. If the connection provider doesn't exist, then the config to start up piezo-admin needs to be updated